### PR TITLE
feat(cli): enable to delete config entry from an input file

### PR DESCRIPTION
**Pull request by huikang**
### Description

- A new flag for `config delete` to delete a config entry in a
  valid config file, e.g., `config delete -filename
  intention-allow.hcl`
- Updated flag validation; -filename and -kind can't be set
  at the same time
- Move decode config entry method from config_write.go to
  helpers.go for reusing ParseConfigEntry()

### Links

#12397 

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

File: website/content/commands/config/delete.mdx [hunk 1]
```diff
@@ -56,6 +56,8 @@ Usage: `consul config delete [options]`
   `proxy-defaults` config entry must be `global`, and the name of the `mesh`
   config entry must be `mesh`.
 
+- `-filename` - Specifies the file of the config entry to delete.
```
Comment (boxofrad) [inline website/content/commands/config/delete.mdx:59]:
```suggestion
- `-filename` - Specifies the file describing the config entry to delete.
```

File: command/config/delete/config_delete.go [hunk 2]
```diff
@@ -115,12 +163,13 @@ func (c *cmd) Help() string {
...
 	help     = `
-Usage: consul config delete [options] -kind <config kind> -name <config name>
+Usage: consul config delete [options] ([-kind <config kind> -name <config name>] | [-f FILENAME])
 
   Deletes the configuration entry specified by the kind and name.
 
   Example:
 
     $ consul config delete -kind service-defaults -name web
+    $ consul config delete -file ./intention-allow-config.hcl
```
[the remark this answers is not in the document]
Comment (boxofrad) [reply command/config/delete/config_delete.go:173]: Also, shall we use the same example filename here as in the docs?

```suggestion
    $ consul config delete -filename service-defaults-web.hcl
```

**Review by boxofrad — APPROVED** [review]
Looks good to me! Left a couple of small suggestions 🙇🏻‍♂️

File: command/config/delete/config_delete_test.go [hunk 3]
```diff
@@ -145,6 +184,18 @@ func TestConfigDelete_InvalidArgs(t *testing.T) {
...
+			args: []string{"-kind", api.ServiceDefaults, "-filename", "config-file.hcl"},
+			err:  "filename can't be used with kind or name",
+		},
+		"name and filename": {
+			args: []string{"-name", "db", "-filename", "config-file.hcl"},
+			err:  "filename can't be used with kind or name",
+		},
+		"kind, name, and filename": {
+			args: []string{"-kind", api.ServiceDefaults, "-name", "db", "-filename", "config-file.hcl"},
+			err:  "Filename and kind,name can't be set together",
```
[the remark this answers is not in the document]
Comment (huikang) [reply command/config/delete/config_delete_test.go]: @boxofrad , agree that these cases should have the same error message. PR is updated. Thanks so much!

**Pull request merged**